vnext/Microsoft.ReactNative/RedBoxHandler.idl

```diff
@@ -3,37 +3,143 @@
 
+#include "DocString.h"
+
 namespace Microsoft.ReactNative {
 
   enum RedBoxErrorType {
+    DOC_STRING("A JS Exception was thrown and not caught or otherwise fatal error")
     JavaScriptFatal, // A JS Exception was thrown or otherwise fatal error
+
+    DOC_STRING("An error coming from JS that isn't fatal, such as `console.error`")
     JavaScriptSoft, // An error coming from JS that isn't fatal, such as console.error
+
+    DOC_STRING("An error happened in native code")
     Native,
   };
 
-  [webhosthidden] interface IRedBoxErrorFrameInfo {
+  [webhosthidden]
+  DOC_STRING("This object represents a single frame within the call stack of an error.")
+  interface IRedBoxErrorFrameInfo {
+    DOC_STRING("The file location of this frame")
     String File { get; };
+
+    DOC_STRING("The method name of this frame")
     String Method { get; };
+
+    DOC_STRING("The line number within the file")
     UInt32 Line { get; };
+
+    DOC_STRING("The column within the line")
     UInt32 Column { get; };
+
+    DOC_STRING("True if this frame is part of the internals of `react-native`, that is likely not useful for the developer to see.")
     Boolean Collapse { get; };
   }
 
-  [webhosthidden] interface IRedBoxErrorInfo {
+  [webhosthidden]
+  DOC_STRING("This object provides information about the error.  For JavaScript errors, a call stack is also provided.")
+  interface IRedBoxErrorInfo {
+    DOC_STRING("The error message.")
     String Message { get; };
+
+    DOC_STRING("If the message was adjusted for formatting, or otherwise processed, this contains the message before those modifications")
     String OriginalMessage { get; };
+
+    DOC_STRING("An identifier for this error.")
     String Name { get; };
+
+    DOC_STRING("This will contain the component stack where the error occurred, which can help identify the component that is producing the error")
     String ComponentStack { get; };
+
+    DOC_STRING("This Id can be used in [`UpdateError`](#updateerror) to identify which error is being updated.  For native errors, this is currently always `0`, and [`UpdateError`](#updateerror) will never be called.")
     UInt32 Id { get; };
+
+    DOC_STRING("For JavaScript errors, this will contain the call stack of where the error occurred.")
     IVectorView<IRedBoxErrorFrameInfo> Callstack { get; };
+
+    DOC_STRING("Provides access to extra data attached to the error.  Adding additional data to the errors is not yet part of the stable API.")
     IJSValueReader ExtraData { get; };
   }
 
   [webhosthidden]
+#ifdef USE_DOCSTRING
+  [doc_string(
+    "`IRedBoxHandler` provides an extension point to allow custom error handling within the React instance.  This can be useful if you have an existing error reporting system that you want React errors to be reported to.  The default implementation of `RedBoxHandler` shows an error messages in a error screen that covers the whole application window.
+
+-- Insert Screenshot here --
```

Comment:
Was this something you wanted to do before pushing, or something intentional?

Reply:
I copied this from the actual docs published on the website :) Someone should add the screenshot :)